erpnext/setup/doctype/territory/territory.js

```diff
@@ -11,29 +11,17 @@ frappe.ui.form.on("Territory", {
 				}
 			}
 		};
+		frm.fields_dict['parent_territory'].get_query = function(doc) {
+			return{
+				filters:[
+					['Territory', 'is_group', '=', 1],
+					['Territory', 'name', "!=", doc.territory_name]
+				]
+			}
+		}
+	},
+	refresh: function(frm) {
+		// read-only for root territory group
+		frm.set_root_read_only("parent_territory");
```

Comment:
- This doesn't seem like a fix for the Treeview, only code is being moved.
- Ask for the roles assigned to users who are facing this issue.
- This issue cannot be replicated with administrator access.

Reply:
@hrwX  its fix for the task [From the List view, I am not able to create a child node in their Parent Node.](https://app.asana.com/0/1192407894480744/1198506200773421)

Where when users try to create a child or root node from the list view.
Had discussed with Swati facing this issue for all role when they try to create a doc from list view